### ISSUE TITLE
Identify connection performance (ping) in-game

### ISSUE THREAD

**Issue by FRenzy-OpenRA** [OpenRA/OpenRA#15834]
It would be useful to identify where the source of a "lag" comes from, while being in-game.

Maybe a ping value for each player and spectator could be an indicator here ? 

Also, should work with "In-game kicking" #3516

**Comment by FRenzy-OpenRA** [comment]
That is true. 

There is still some kind of identification nowadays, and filtering. 
Currently, in case of lag and in general, all spectators are asked to leave one by one (sometimes a player appears to be the cause), and usually they're remembered next games. 

And in any case, kicking often happens before the start with latency values / colors. 


I'd argue that sometimes, a game can go well even if players appear to lag before starting. And vice versa. 
It is punishing for the players, and the potential spectators, not to be able to know where the problem comes from. And after deciding a re-make due to lag, it results in kicking a lot of spectators to avoid potential lag.

**Comment by netnazgul** [comment]
Main problem with kicking potential laggers is that you can't remove them from the game once it is started, so people try to prevent it before. Ingame admin tools would be nice to have, and only then the ingame latency indicator might help.

**Comment by Jehoel** [comment]
"Lag" is more than just packet latency, of course - and games can be ruined by players with inadequate computer performance on the best of Internet connections.

The OpenRA server should be modified to identify which connected clients are the cause of any slowdowns and report them in-game as it does with broken connections.
